keyboards/handwired/uc2/keymaps/via/keymap.c

```diff
@@ -0,0 +1,129 @@
+/* Copyright 2021 AndyChiu
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) { /* First encoder */
+    if (clockwise) {
+      //tap_code(KC_MS_WH_UP);
+      tap_code(dynamic_keymap_get_keycode(biton32(layer_state),0,7));
+    } else {
+      //tap_code(KC_MS_WH_DOWN);
+      tap_code(dynamic_keymap_get_keycode(biton32(layer_state),1,7));
+    }
+  }else if (index == 1) { /* First encoder */
+    if (clockwise) {
+      //tap_code(KC_MS_WH_UP);
+      tap_code(dynamic_keymap_get_keycode(biton32(layer_state),2,7));
+    } else {
+      //tap_code(KC_MS_WH_DOWN);
+      tap_code(dynamic_keymap_get_keycode(biton32(layer_state),3,7));
+    }
+  }
+  return true;
+  }
```

Comment:
This won't actually work correctly, as tap_code (and tap_code16) only support basic keycodes (and modded basic keycodes). 

Ideally, this should be using the Encoder Map feature, but that's not supported by VIA yet. Though, the dynamic keymap feature (which is what powers VIA, really) actively supports it. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

In the meanwhile, something like this should be used:
https://github.com/qmk/qmk_firmware/blob/b66b911fa20debfb6e49756dee57d22228f5f122/keyboards/dailycraft/stickey4/keymaps/via/keymap.c#L45-L115

This allows for proper handling of all keycodes, without special cases.